fbpic/print_utils.py

```diff
@@ -0,0 +1,234 @@
+                if threading_enabled:
+                    message += '\nCPU multi-threading enabled: Yes'
+                    message += '\nThreads: %s' \
+                        %numba.config.NUMBA_NUM_THREADS
+                else:
+                    message += '\nCPU multi-threading enabled: No'
+                if mkl_installed:
+                    message += '\nFFT library: MKL'
+                else:
+                    message += '\nFFT library: pyFFTW'
+
+            message += '\n'
+            if sim.fld.n_order == -1:
+                message += '\nPSAOTD stencil order: infinite'
+            else:
+                message += '\nPSAOTD stencil order: %d' \
+                    %sim.fld.n_order
+            message += '\nParticle shape: %s' %sim.particle_shape
+            message += '\nLongitudinal boundaries: %s' %sim.comm.boundaries
+            message += '\nTransverse boundaries: reflective'
+            message += '\nGuard region size: %d ' \
+                %sim.comm.n_guard + 'cells'
+            message += '\nDamping region size: %d ' \
+                %sim.comm.n_damp + 'cells'
+            message += '\nParticle exchange period: every %d ' \
+                %sim.comm.exchange_period + 'step'
+            if sim.gamma_boost is not None:
+                message += '\nBoosted frame: Yes'
+                message += '\nBoosted frame gamma: %d' \
+                    %sim.comm.gamma_boost
+                if sim.use_galilean:
+                    message += '\nGalilean frame: Yes'
+                else:
+                    message += '\nGalilean frame: No'
+            else:
+                message += '\nBoosted frame: False'
+        message += '\n'
+        print( message )
+    if level == 2:
+        # Sync MPI processes before printing MPI GPU selection
+        sim.comm.mpi_comm.barrier()
+        time.sleep(0.1)
+        if sim.use_cuda:
+            print_current_gpu( MPI )
+            sim.comm.mpi_comm.barrier()
+            time.sleep(0.1)
+            if sim.comm.rank == 0:
+                print('')
+
+def progression_bar( i, Ntot, avg_time_per_step, prev_time,
+                     n_avg=20, Nbars=35, char=progress_char):
+    """
+    Shows a progression bar with Nbars and the estimated
+    remaining simulation time.
+    """
+    # Estimate average time per step
+    curr_time = time.time()
+    time_per_step = curr_time - prev_time
+    avg_time_per_step += (time_per_step - avg_time_per_step)/n_avg
+    if i <= 2:
+        # Ignores first step in time estimation (compilation time)
+        avg_time_per_step = time_per_step
+    # Print progress bar
+    if i == 0:
+        # Let the user know that the first step is much longer
+        sys.stdout.write('\r' + \
+            'Just-In-Time compilation (up to one minute) ...')
+        sys.stdout.flush()
+    else:
+        # Print the progression bar
+        nbars = int( (i+1)*1./Ntot*Nbars )
+        sys.stdout.write('\r' + nbars*char )
```

Comment:
I have a slight personal preference for having the square brackets around the nbars characters (esp. when the character is the simple '-'), like this:
```
[--------             ]
```
Any strong opinion on this?

Reply:
The new default style of the progress bar does not look nice together with the brackets. (I hope you like the new default style). Therefore, I would remove the brackets.

Another option would be to use " | " as an indicator for the start and end of the progress bar. This looks okay with the new default style.